### ISSUE TITLE
Recordedit sends invalid floating point value

### ISSUE THREAD

**Issue by karlcz** [informatics-isi-edu/chaise#477]
Entering data for a table with a floating point column, I can get the UI to send an invalid empty string literal `""` instead of a `null` when the entry field is empty.

1. Fill in a valid value such as `12`
2. Click the duplicate last record button
3. Edit the field in the new (duplicated) record to delete all the characters
4. Press submit

Instead of sending a `null` in the JSON input, it sends an empty string.  This causes an error on the server since an empty string cannot be converted to a floating point number. I think the application should always send a `null` when a text-entry field is empty, regardless of column type in the table definition.  If we ever get an obscure scenario where we need to enter a literal empty string value, we can discuss the UX for that case at that time.

**Comment by jrchudy** [comment]
I tested this on both Chrome and Firefox and could not reproduce the issue. When submitting the data after removing the value in the duplicated row, it's submitting that value as `null`. @karlcz are you able to still reproduce this or am I missing a step?

**Comment by karlcz** [comment]
I just repeated the error on synapse-dev using firefox.

Follow my steps **exactly**:

1. Goto https://synapse-dev.isrd.isi.edu/chaise/recordset/Zebfrafish:Synapse and login
2. Goto https://synapse-dev.isrd.isi.edu/chaise/recordedit/#2/Zebrafish:Subject
3. Fill in all required columns and put `12` in the RNA_Conc and DNA_Conc columns
4. Click DNA_Conc field to put cursor/focus at end of line and backspace over `12` so it is empty
5. Click submit button
6. See error message instead of success

It doesn't actually require the duplicate record feature at all...

The displayed error is what I said before.  It is sending `""` instead
of `null` to the server for the `DNA_Conc` column.

**Comment by jrchudy** [comment]
This issue is addressed in my most recent commit. This can be tested [here](https://synapse-dev.isrd.isi.edu/~jchudy/chaise/recordedit/#2/Zebrafish:Subject) in my synapse dev environment.

This issue was caused by columns with data type `float4` and `float8`. It wasn't explicitly spec'ed out [here](https://github.com/informatics-isi-edu/division/issues/35) when we came up with the set of data types for `recordedit` so those types were defaulting to a plain `text` input field. Text inputs, by default, submit `''` instead of `null`. If we had set a `numeric` input for these float types, this would not have been an issue.

Assigning to @karlcz for review. Changes are in this [commit](https://github.com/informatics-isi-edu/chaise/commit/2f188f07de8e434df197a6e0cf81b3dbc89c207a).
